addons/web/static/src/js/services/notification_service.js

```diff
@@ -82,8 +85,25 @@ var NotificationService = AbstractService.extend({
         }
         var NotificationWidget = params.Notification || Notification;
         var notification = this.notifications[++id] = new NotificationWidget(this, params);
-        notification.appendTo(this.$el);
-        return id;
+        if (params.await) {
+            return notification.appendTo(this.$el).then(function () {
+                return id;
+            });
+        } else {
+            notification.appendTo(this.$el);
+            return id;
+        }
```

Comment:
Meh feels kinda bad to have a function with different return types... problematic to always return a promise I guess?

Reply:
Yep it breaks the web tests